src/main/java/io/stargate/sgv2/jsonapi/service/updater/DocumentUpdater.java

```diff
@@ -70,6 +87,29 @@ private boolean update(ObjectNode docToUpdate, boolean docInserted) {
     return modified;
   }
 
+  /**
+   * vectorize UpdateClause as needed, only when documents are found or upsert will be used by
+   * updateOne, findOneAndUpdate, updateMany
+   *
+   * @param dataVectorizer
+   */
+  public void vectorizeUpdateClause(DataVectorizer dataVectorizer) {
+    try {
+      dataVectorizer
+          .vectorizeUpdateClause(updateClause)
+          .runSubscriptionOn(Infrastructure.getDefaultWorkerPool())
+          .subscribeAsCompletionStage()
+          .get();
+    } catch (Exception e) {
+      if (e instanceof ExecutionException exception) {
+        if (exception.getCause() instanceof JsonApiException jsonApiException) {
+          throw jsonApiException;
+        }
+      }
+      throw new RuntimeException(e);
```

Comment:
minor: should not wrap if already a `RuntimeException`.

Reply:
deleted, thanks